### PR TITLE
chore(ci): delete stale visual snapshot branches workflow

### PR DESCRIPTION
**Pull request by dtsanevmw**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

Deletes stale visual snapshot branches on manual trigger.

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [X] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Not deleting them anyway.

Issue Number: CDE-1562

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: .github/workflows/delete-stale-branches.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Delete stale visual snapshot branches
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+
+permissions:
+  issues: write
```
Comment (kevinbuhmann) [inline .github/workflows/delete-stale-branches.yml:9]: Why is this permission needed?

Comment (dtsanevmw) [reply .github/workflows/delete-stale-branches.yml:9]: It's supposed to create an issue before that stale branch is deleted but I suspect we don't need that for visual snapshots.

Comment (dtsanevmw) [reply .github/workflows/delete-stale-branches.yml]: I used a different action that is not creating issues and we would trigger manually. I don't want to run on daily schedule just for better safety.

File: .github/workflows/delete-stale-branches-preview.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        uses: phpdocker-io/github-actions-delete-abandoned-branches@v2
+        with:
+          github_token: ${{ github.token }}
+          last_commit_age_days: 120
+          allowed_prefixes: visual-snapshot-update/pr-
+          ignore_branches: main,next,beta,15.x,13.x
+          # Disable dry run and actually get stuff deleted
+          dry_run: yes
+      - name: Get output
+        run: "echo 'Deleted branches: ${{ steps.delete_stuff.outputs.deleted_branches }}'"
```
Comment (kevinbuhmann) [inline .github/workflows/delete-stale-branches-preview.yml:20]: The "Delete visual snapshot branches" step needs `id: delete_stuff` for this to work.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]

File: .github/workflows/delete-stale-branches-preview.yml (hunk 2 above)
Comment (kevinbuhmann) [reply .github/workflows/delete-stale-branches-preview.yml:20]: I would name it `delete_branches` though.

File: .github/workflows/delete-stale-branches.yml (hunk 1 above)
Comment (kevinbuhmann) [reply .github/workflows/delete-stale-branches.yml]: Nice! Thanks!

File: .github/workflows/delete-stale-branches-preview.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  cleanup_visual_snapshot_branches:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Delete visual snapshot branches
+        uses: phpdocker-io/github-actions-delete-abandoned-branches@v2
+        with:
+          github_token: ${{ github.token }}
+          last_commit_age_days: 120
+          allowed_prefixes: visual-snapshot-update/pr-
+          ignore_branches: main,next,beta,15.x,13.x
```
Comment (kevinbuhmann) [inline .github/workflows/delete-stale-branches-preview.yml:16]: Is this needed? These branches don't match the `allowed_prefixes` anyway so this seems like something we don't need to maintain. If it's not needed, I think it should just be removed for simplicity.

File: .github/workflows/delete-stale-branches-preview.yml (hunk 2 above)
Comment (dtsanevmw) [reply .github/workflows/delete-stale-branches-preview.yml:20]: Yes, my mistake...fixed it 👍

File: .github/workflows/delete-stale-branches-preview.yml (hunk 3 above)
Comment (kevinbuhmann) [reply .github/workflows/delete-stale-branches-preview.yml:17]: Okay, that's fine.

**Review by kevinbuhmann — APPROVED** [review]

**Comment by kevinbuhmann** [comment]
I think I see the issue with the existing workflows. I am going to open a PR to fix it. We might not need this.

**Comment by kevinbuhmann** [comment]
@dtsanevmw: I think #1129 solves the root issue. These workflows could still be useful though for branches that slip through if PR builds are canceled for whatever reason (e.g. PR close).

File: .github/workflows/delete-stale-branches-preview.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+name: Delete stale visual snapshot branches Preview
```
Comment (kevinbuhmann) [inline .github/workflows/delete-stale-branches-preview.yml:1]:
```suggestion
name: Delete Stale Visual Snapshot Branches Preview
```

File: .github/workflows/delete-stale-branches.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+name: Delete stale visual snapshot branches
```
Comment (kevinbuhmann) [inline .github/workflows/delete-stale-branches.yml:1]:
```suggestion
name: Delete Stale Visual Snapshot Branches
```

**Review by kevinbuhmann — COMMENTED** [review]
I would rename the workflows using title case to match the other workflows in the list.

**Pull request merged**